ci/bamboo/build-windows.ps1

```diff
@@ -0,0 +1,227 @@
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+# Builds and tests the nupic.bindings python extension using Release build
+# type.
+#
+# ASUMPTIONS:
+#          1. Building via Numenta Vagrant Task using Vagrant image
+#             bamboo-vagrant-windows-1.x.x.
+#
+#          2. nupic.core root is the current directory on entry.
+#
+#          3. Expects a pristine nupic.core source tree without any remnant
+#             build artifacts from prior build attempts. Otherwise, the
+#             behavior is undefined.
+#
+# NOTES: Based on examples in comments of
+#        https://github.com/NumentaCorp/infrastructure/pull/376
+#
+# OUTPUTS:
+#   nupic.bindings wheel: On success, the resulting wheel will be located in
+#                         the subdirectory nupic_bindings_wheelhouse of the
+#                         source tree's root directory.
+#
+#   Test results: nupic.bindings test results will be located in the
+#                 subdirectory test_results of the source tree's root directory
+#                 with the following content:
+#
+#                 junit-test-results.xml
+#                 htmlcov/
+
+
+# Stop and fail script if any command fails
+$ErrorActionPreference = "Stop"
+
+
+$NupicCoreRootDir = $(get-location).Path
+
+
+
+# Use this function to wrap external commands for powershell error-checking.
+#
+# This is necessary so that `$ErrorActionPreference = "Stop"` will have the
+# desired effect.
+#
+# Returns True if command's $LastExitCode was 0, False otherwise
+#
+# Usage: WrapCmd { cmd arg1 arg2 ... }
+#
+function WrapCmd
+{
+  [CmdletBinding()]
+
+  param (
+    [Parameter(Position=0, Mandatory=1)]
+    [scriptblock]$Command,
+    [Parameter(Position=1, Mandatory=0)]
+    [string]$ErrorMessage = "ERROR: Command failed.`n$Command"
+  )
+  & $Command
+  if ($LastExitCode -eq 0) {
+    return $true
+  }
+  else {
+    Write-Error "WrapCmd: $ErrorMessage"
+    return $false
+  }
+}
+
+
+#
+# Make unix-compatible patch.exe available to the build by copying it from
+# Git\usr\bin to another directory and adding it to PATH; the reason we copy it
+# is that Git\usr\bin contains sh.exe that cmake doesn't like.
+#
+
+# Verify that patch command is not available yet.
+&where.exe patch
+if ($LastExitCode -eq 0) {
+    throw "patch command was already available."
+}
+
+mkdir "C:\Program Files\PatchFromGit"
+copy "C:\Program Files\Git\usr\bin\patch.exe" "C:\Program Files\PatchFromGit"
+copy "C:\Program Files\Git\usr\bin\msys*.dll" "C:\Program Files\PatchFromGit"
+$env:PATH = 'C:\Program Files\PatchFromGit;' + $env:PATH
+
+# Verify that patch is now available
+WrapCmd { where.exe patch }
+
+
+#
+# Remove sh.exe from the paths (CMake doesn't like it and fails in a subtle way)
```

Comment:
Hmm, I'm not really sure what the best way to handle it is. I'm fine with how it is for now.